addons/events/src/components/Event.tsx

```diff
@@ -98,12 +121,17 @@ class Item extends Component {
     payload: {},
   };
 
-  state = {
+  state: ItemState = {
     isTextAreaShowed: false,
+    failed: false,
+    payload: null,
+    payloadString: '',
+    // eslint-disable-next-line react/no-unused-state,
+    prevPayload: null,
   };
 
-  onChange = ({ target: { value } }) => {
-    const newState = {
+  onChange = ({ target: { value } }: React.ChangeEvent<HTMLTextAreaElement>) => {
+    const newState: Partial<ItemState> = {
```

Comment:
Just another newbie question... What is the right way to initialize an object in typescript? I've read [this](https://basarat.gitbooks.io/typescript/docs/tips/lazyObjectLiteralInitialization.html) and articles on google in general... but still I'm not sure. I've used `Partial` because the current setting is `no-object-literal-type-assertion`. Not sure below's `this.setState(newState as ItemState)` too

Reply:
I think partial is fine! However, you should remove `newState as ItemState`

Reply:
@wuweiweiwu removing `newState as ItemState` will cause TS error 😞 

A workaround is to do something like:
```ts
try {
    this.setState({
        payloadString: value,
        payload: JSON.parse(value.trim()),
        failed: false
    });
} catch (err) {
    this.setState({
        payloadString: value,
        failed: true
    });
}
```
But I'm not really convinced...

Reply:
@gaetanmaisse yeah I'll try to find something more elegant(?) haha

Reply:
I tried some other ways but this one looks better to me. In this example, `state` is `Readonly<ItemState>` and `newState` is `Partial<ItemState>` 
```
this.setState(state => ({ ...state, ...newState }));
```